### DISCUSSION TITLE
Running commands in HOOKS

### DISCUSSION THREAD

**Discussion by robflate** [garethgeorge/backrest#484] [Q&A] [answered]
I need a little guidance on running commands in `HOOKS`. The following gives an error;

`docker ps --format '{{.Names}}' | grep -v 'backrest|traefik|crowdsec|socket-proxy' | xargs docker stop`

**Error**

`error: fatal: template rendering: execute template: template: template:1:22: executing "template" at <.Names>: can't evaluate field Names in type tasks.HookVars`

I'd rather have the commands in the GUI rather than making separate scripts. Advice and general guidance on how to make commands suitable for Backrest appreciated, thanks.

**Comment by garethgeorge** [comment, marked as answer]
Ah, this is a bit tricky. Backrest uses go templates for substituting variables into scripts and messages, and so does docker cli :)

You'll need to escape the go template string `{{ .Names }}` e.g. as `{{"{{"}} .Names {{"}}"}}` (which yes -- is pretty ugly).
